modules/geo-layers/src/wms-layer/wms-layer.ts

```diff
@@ -244,6 +252,26 @@ export class WMSLayer<ExtraPropsT extends {} = {}> extends CompositeLayer<
     }
   }
 
+  /** Checks if an image request would be successful */
+  protected canRequestImage(viewport: Viewport, props: WMSLayerProps): boolean {
+    // WMS services require at least one "data layer" to be specified
+    const {layers, serviceType} = this.props;
+    if (serviceType === 'wms' && layers.length === 0) {
+      return false;
+    }
+
+    // Bounds must not be 0,0,0,0
+    const bounds = viewport.getBounds();
+    const {width, height} = viewport;
+
+    const validSizeAndBounds =
+      width > 0 &&
+      height > 0 &&
+      !(bounds[0] === 0 && bounds[1] === 0 && bounds[2] === 0 && bounds[3] === 0);
```

Comment:
This seems oddly specific to me. Does it only fail when all four numbers are 0, or whenever the requested area is 0?

Reply:
I have only seen WMS services emit errors when all are 0, and this seems to happen when the layer first renders, then the viewport immediately gets proper values and it is no longer an issue. 

I assume that boundingBox can never become zero area in steady-state use?

Reply:
It can, e.g. if the container has 0px height.

Reply:
Then we get height=0 which triggers early return?